server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -300,11 +300,20 @@ export default class ServiceProviderReferralsController {
     const form = await AddActionPlanActivitiesForm.createForm(req)
 
     if (form.isValid) {
-      await this.interventionsService.updateDraftActionPlan(
-        res.locals.user.token.accessToken,
-        req.params.id,
-        form.activityParamsForUpdate
-      )
+      if (form.activityParamsForUpdate.id) {
```

Comment:
would be good to abstract this into a method e.g. `shouldUpdateExistingActivity`

Reply:
there should be a test for this new code.

Reply:
done